### PR TITLE
[Sprite Lab] Warnings for pointer blocks

### PR DESCRIPTION
**Pull request by mikeharv**
Blockly has a Warning Indicator system that can alert a student when they are using a block in an unexpected way.

https://blocklycodelabs.dev/codelabs/validation-and-warnings/index.html#0

This is implemented in Core for two blocks, neither of which we use:

image (220)

After discovering the feature in July, I discussed possible benefits to Sprite Lab with the then-K5 team.

Sprite Lab has some "pointer" blocks that meant to work exclusively with certain event blocks. If we used warnings when these blocks are misplaced, it could give students (or their teachers) the nudge they need to get unstuck.

Katie confirmed (via DM) that this solution was a desired improvement, and it was easy enough to implement.

Expected placement (no warning):
image

Unexpected placement (under "when run") adds a clickable warning icon:
image

If the block is disabled on the workspace, no warning is shown:
image

The `block.setWarningText` [method](https://developers.google.com/blockly/reference/js/blockly.block_class.setwarningtext_1_method) takes either a string or `null`(to delete the warning). We call this method whenever the block changes, in order to keep it up to date. (See Testing section for a longer video.)

I added this logic to an existing custom `onchange` function, which already handles block image "shadowing" for these pointer blocks. The `onBlockImageSourceChange` function uses a `spriteLabPointers` object which maps block types. A block's `imageSourceType` refers to the expected root (top-most) block, so I was able to reuse this map. I renamed the properties to better reflect its purpose for these two separate uses.

## Links

Slack thread: https://codedotorg.slack.com/archives/C060J3A3BBK/p1722003871913629

## Testing story

Manually tested

https://github.com/user-attachments/assets/fe7c5067-895f-43fa-aa57-e2c170f12df1

## Follow-up work

In addition to the benefitting our Sprite Lab users, I have a theory that this system could come in handy for Music Lab's CSD3 re-write using Advanced mode. If we decide to do more with the feature, this serves as a nice first practice step.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -1430,7 +1433,6 @@
   "incubator_aiRubric_descLink": "\"This AI Grading Tool Cut One Teacher's Grading Time in Half. Here's How it Works\".",
   "incubator_aiRubric_desc2": "Apply for the pilot and get help assessing student projects with our new AI Teaching Assistant. Available in CSD Unit 3 - Interactive Games and Animations.",
   "incubator_aiRubricLearnMore_button": "Learn More",
-  "incubator_canvasIntegration_earlyAccess_guides_button": "Guides",
```
Comment (molly-moen) [inline apps/i18n/common/en_us.json:1433]: did you mean to delete this?

**Review by molly-moen — COMMENTED** [review]
This is a cool feature! One question on a deleted string

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (mikeharv) [reply apps/i18n/common/en_us.json:1433]: Oh, yes! Meant to leave an in-line comment, but I saw this was an exact duplicate on line 1427 (old) / 1430 (new). Seemed safe to clean up.

Comment (molly-moen) [reply apps/i18n/common/en_us.json:1433]: 👍

**Review by molly-moen — APPROVED** [review]
Nice!

**Pull request merged**